frontend/src/context/user-context.tsx

```diff
@@ -30,13 +31,14 @@ export type UserProviderProps = {
 
 export const UserProvider = component$((initialUser: UserProviderProps) => {
   const user = useStore<User>(initialUser.user);
+  const nav = useNavigate();
 
   // eslint-disable-next-line qwik/no-use-visible-task
   useVisibleTask$(async () => {
     const existingJsonUser = localStorage.getItem('user');
 
     if ( existingJsonUser === null ) {
-      localStorage.setItem('user', JSON.stringify(user));
+      nav('/login');
```

Comment:
Navigation inside a `...Provider` is a bit hidden. We can expose a new function `isLogged` from the context provider and when the output of this `fn` is `false`, we force the navigation from the outer component. What do you think?